src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -1353,7 +1353,7 @@ public void visitClassDef(JCClassDecl tree) {
         public void visitIdent(JCIdent tree) {
             if (context() != null && lambdaIdentSymbolFilter(tree.sym)) {
                 if (tree.sym.kind == VAR &&
-                        tree.sym.owner.kind == MTH &&
+                        (tree.sym.owner.kind == MTH || tree.sym.owner.kind == VAR) &&
```

Comment:
just curious: why wasn't this code stressed before? I mean it should be possible to stress it without switch expressions right?

Reply:
Uhm - don't we always try to create a block/synthetic method symbol for owner of variables declared inside initializers? I think the case where this happens is lambda expression parameters - which can occur in field initializers. The parameters should point to some "init block" as their owner, not the var. Shouldn't switch behave the same?